misc_commands_test.go

```diff
@@ -87,143 +57,408 @@ func TestParseRpbGetBucketRespCorrectly(t *testing.T) {
+		if got, want := fetchBucketTypePropsCommand.success, true; got != want {
+			t.Errorf("got %v, want %v", got, want)
 		}
-		if expected, actual := "module_name", response.PostCommit[0].ModFun.Module; expected != actual {
-			t.Errorf("expected %v, actual %v", expected, actual)
+		validateFetchBucketPropsResponse(t, fetchBucketTypePropsCommand.Response)
+	} else {
+		t.Errorf("ok: %v - could not convert %v to *FetchBucketTypePropsCommand", ok, reflect.TypeOf(cmd))
+	}
+}
+
+// FetchBucketProps
+
+func TestBuildRpbGetBucketReqCorrectlyViaBuilder(t *testing.T) {
+	builder := NewFetchBucketPropsCommandBuilder().
+		WithBucketType("bucket_type").
+		WithBucket("bucket_name")
+	cmd, err := builder.Build()
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	protobuf, err := cmd.constructPbRequest()
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	if protobuf == nil {
+		t.Fatal("protobuf is nil")
+	}
+	if req, ok := protobuf.(*rpbRiak.RpbGetBucketReq); ok {
+		if got, want := string(req.GetType()), "bucket_type"; got != want {
+			t.Errorf("got %v, want %v", got, want)
 		}
-		if expected, actual := "function_name", response.PostCommit[0].ModFun.Function; expected != actual {
-			t.Errorf("expected %v, actual %v", expected, actual)
+		if got, want := string(req.GetBucket()), "bucket_name"; got != want {
+			t.Errorf("got %v, want %v", got, want)
 		}
-		if expected, actual := "module_name", response.ChashKeyFun.Module; expected != actual {
-			t.Errorf("expected %v, actual %v", expected, actual)
+	} else {
+		t.Errorf("ok: %v - could not convert %v to *rpbRiak.RpbGetBucketReq", ok, reflect.TypeOf(protobuf))
+	}
+}
+
+func TestParseRpbGetBucketRespCorrectly(t *testing.T) {
+	builder := NewFetchBucketPropsCommandBuilder()
+	cmd, err := builder.
+		WithBucketType("bucket_type").
+		WithBucket("bucket_name").
+		Build()
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	err = cmd.onSuccess(buildRpbGetBucketResp())
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	if got, want := cmd.Success(), true; got != want {
+		t.Errorf("got %v, want %v", got, want)
+	}
+
+	if fetchBucketPropsCommand, ok := cmd.(*FetchBucketPropsCommand); ok {
+		if fetchBucketPropsCommand.Response == nil {
+			t.Fatal("unexpected nil object")
 		}
-		if expected, actual := "function_name", response.ChashKeyFun.Function; expected != actual {
-			t.Errorf("expected %v, actual %v", expected, actual)
+		if got, want := fetchBucketPropsCommand.success, true; got != want {
+			t.Errorf("got %v, want %v", got, want)
 		}
-		if expected, actual := "module_name", response.LinkFun.Module; expected != actual {
-			t.Errorf("expected %v, actual %v", expected, actual)
+		validateFetchBucketPropsResponse(t, fetchBucketPropsCommand.Response)
+	} else {
+		t.Errorf("ok: %v - could not convert %v to *FetchBucketPropsCommand", ok, reflect.TypeOf(cmd))
+	}
+}
+
+// StoreBucketTypeProps
+
+func TestBuildRpbSetBucketTypeReqCorrectlyViaBuilder(t *testing.T) {
```

Comment:
Did you mean Store instead of Set for the test name? Seems like you use Store everywhere else.